utils/build/build.mk

```diff
@@ -65,13 +73,21 @@ MAIN_OBJ  := $(patsubst run/%.cpp, $(OBJS_DIR)/%.o, $(MAIN))
 $(TARGET): $(DEPENDENCIES) $(OBJS) $(MAIN_OBJ)
 	$(Echo) "Linking executable $(MAIN) into $@"
 	$(Verb) $(LL)  -o $@ $(OBJS) $(MAIN_OBJ) $(LFLAGS) $(COVERAGE_FLAGS)
+ifeq ($(USING_CLANG), 1)
+# This finds all the individual compilation database files in the bin directory,
+# and combines them into one compile_commands.json file
+# Only works with gnu sed, couldn't get newlines to work with bsd sed
+# On mac use homebrew to install gnu sed
+# Double $ signs are so Make interprets it as a single regex $, on command line use a single $ sign
+	$(Verb) find bin -name '*.json' -exec sed -e '1s/^/[\n/' -e '$$s/,$$/\n]/' {} + > compile_commands.json
```

Comment:
Could you clean the temporary `.json` files once you do not need them?

Reply:
Sure, this should be possible, but I'm not quite sure why this would be beneficial? The `.json` files are in `bin` anyways, would there be a difference?